### PR TITLE
Amir - intagrated the move to pose to the scoring positions

### PR DESCRIPTION
**Pull request by YoniKiriaty**

### CODE REVIEW

File: src/main/java/edu/greenblitz/robotName/Field.java [hunk 1]
```diff
@@ -116,4 +95,10 @@ public static class FieldConstants{
         public final static double fieldLength = 16.54175;
         public final static double fieldWidth = 8.0137;
     }
+
+    public static class ScoringPositions{
+        public final static Pose2d speakerScorePosition = new Pose2d(new Translation2d(5,4),new Rotation2d());
+        public final static Pose2d ampScorePosition = new Pose2d(new Translation2d(3,1),new Rotation2d());
+        public final static Pose2d climbPosition = new Pose2d(new Translation2d(7,2),new Rotation2d());
```
Comment (tnrz135) [inline src/main/java/edu/greenblitz/robotName/Field.java:102]: Those are consts, they should be UPPER_SNAKE_CASE

**Review by tnrz135 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/swerve/MoveToAmp.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+import edu.greenblitz.robotName.Field;
+import edu.greenblitz.robotName.commands.auto.MoveToPose;
+
+public class MoveToAmp extends SwerveCommand{
+    @Override
+    public void initialize() {
+        MoveToPose.getPathCommand(Field.ScoringPositions.ampScorePosition).schedule();
+    }
+}
```
Comment (aaa-png) [inline src/main/java/edu/greenblitz/robotName/commands/swerve/MoveToAmp.java:11]: Yoni - scheduling in the initialize of a command is very ugly, you can probably change MoveToPose so you can extend it to move to the amp position.

File: src/main/java/edu/greenblitz/robotName/commands/swerve/MoveToClimbPosition.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import edu.greenblitz.robotName.Field;
+import edu.greenblitz.robotName.commands.auto.MoveToPose;
+
+public class MoveToClimbPosition extends SwerveCommand{
+
+    @Override
+    public void initialize() {
+        MoveToPose.getPathCommand(Field.ScoringPositions.climbPosition).schedule();
+    }
+}
```
Comment (aaa-png) [inline src/main/java/edu/greenblitz/robotName/commands/swerve/MoveToClimbPosition.java:12]: Yoni - same here

File: src/main/java/edu/greenblitz/robotName/commands/swerve/MoveToSpeaker.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+
+import edu.greenblitz.robotName.Field;
+import edu.greenblitz.robotName.commands.auto.MoveToPose;
+
+public class MoveToSpeaker extends SwerveCommand {
+    @Override
+    public void initialize() {
+        MoveToPose.getPathCommand(Field.ScoringPositions.speakerScorePosition).schedule();
+    }
+}
```
Comment (aaa-png) [inline src/main/java/edu/greenblitz/robotName/commands/swerve/MoveToSpeaker.java:11]: Yoni - triple same here

**Review by aaa-png — CHANGES_REQUESTED** [review]

**Review by greenblitz4590 — APPROVED** [review]

File: src/main/java/edu/greenblitz/robotName/commands/auto/MoveToPose.java [hunk 5]
```diff
@@ -3,18 +3,42 @@
...
 import com.pathplanner.lib.path.PathConstraints;
 import edu.greenblitz.robotName.subsystems.swerve.Chassis.ChassisConstants;
+import edu.greenblitz.robotName.utils.GBCommand;
 import edu.wpi.first.math.geometry.Pose2d;
 import edu.wpi.first.wpilibj2.command.Command;
 
-public class MoveToPose {
-    public static Command getPathCommand(Pose2d endPoint) {
-        return getPathCommand(endPoint, ChassisConstants.CONSTRAINTS);
+public class MoveToPose extends GBCommand {
```
Comment (tnrz135) [inline src/main/java/edu/greenblitz/robotName/commands/auto/MoveToPose.java:10]: Pose (stylized as POSE) is an American drama television series about New York City's ball culture, an LGBTQ subculture in the African-American and Latino communities, throughout the 1980s and 1990s. Featured characters are dancers and models, who compete for trophies and recognition in this underground culture and who support one another in a network of chosen families known as Houses.

**Review by tnrz135 — CHANGES_REQUESTED** [review]

**Pull request merged**